lib/service/CsrfService.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+
+namespace CsrDelft\service;
+
+
+use Symfony\Component\Security\Csrf\CsrfToken;
+use Symfony\Component\Security\Csrf\CsrfTokenManager;
+use Symfony\Component\Security\Csrf\CsrfTokenManagerInterface;
+
+class CsrfService {
+	private static $instance;
+	static function instance() : CsrfService {
+		if (static::$instance === null)
+			static::$instance = static::init();
+		return static::$instance;
+	}
+	/**
+	 * CsrfService constructor.
+	 * @param $manager CsrfTokenManagerInterface
+	 */
+	public function __construct($manager) {
+		$this->manager = $manager;
+	}
+	/**
+	 * @var CsrfTokenManagerInterface
+	 */
+	private $manager;
+
+	private static function init() : CsrfService {
+		return new CsrfService(new CsrfTokenManager());
+	}
+
+	/**
+	 * @param $token
+	 * @param string $path
+	 * @param string $method
+	 * @return CsrfToken|null
+	 */
+	public function isValid ($token, string $path, string $method) : bool {
```

Comment:
Wat doen `method` en `path`?

Reply:
Nog niets. Ik ben nog aan het twijfelen of ik het CSRF-token bij formulieren laat afhangen van de doel-url en HTTP-method. Dat zou in bepaalde gevallen extra veiligheid kunnen bieden.